game/plugin/skills/runecrafting/src/runecrafting.plugin.kts

```diff
@@ -0,0 +1,92 @@
+
+import org.apollo.game.message.impl.*
+import org.apollo.game.model.entity.EquipmentConstants
+import org.apollo.game.model.event.impl.LoginEvent
+
+
+on_player_event {LoginEvent::class}
+        .then {
+            val hat = player.equipment.get(EquipmentConstants.HAT)
+            if (hat != null) {
+                val tiara = Tiara.findById(hat.id)
+                if (tiara != null) {
+                    player.send(ConfigMessage(CHANGE_ALTER_OBJECT_CONFIG, 1 shl tiara.bitshift))
+                } else {
+                    player.send(ConfigMessage(CHANGE_ALTER_OBJECT_CONFIG, 0))
+                }
+            } else {
+                player.send(ConfigMessage(CHANGE_ALTER_OBJECT_CONFIG, 0))
+            }
+        }
+
+on {ObjectActionMessage::class}
+        .where {option == 2}
+        .then {
+            val taria = Tiara.findByAlterId(id)
+            //Check if player has taria
+            val hat = it.equipment.get(EquipmentConstants.HAT)
+            if (taria != null && hat != null && hat.id == taria.id) {
+                //Check if correct alter
+                if (taria.alter.entranceId == id) {
+                    it.startAction(TeleportAction(it, position, 2, taria.alter.entrance))
+                    terminate()
+                }
+            }
+        }
+
+on {ItemActionMessage::class}
+        .where { option == 1 }
+        .then {
+            val taria = Tiara.findById(id)
+            if (taria != null) {
+                it.send(ConfigMessage(CHANGE_ALTER_OBJECT_CONFIG, 0))
```

Comment:
Would be good to put this in a utility function somewhere with a meaningful name.  I'm not sure what it does at the moment.